test/worker_test.py

```diff
@@ -1296,3 +1298,358 @@ def store_task(t, error_msg):
         w._handle_next_task()
 
         self.assertEqual(result, [task])
+
+
+class PerTaskRetryPolicyBehaviorTest(unittest.TestCase):
+    def setUp(self):
+        self.per_task_retry_count = 2
+        self.default_retry_count = 1
+        self.sch = Scheduler(retry_delay=0.1, retry_count=self.default_retry_count, prune_on_get_work=True, record_task_history=False)
+
+        self.test_instance_id = str(uuid4())
```

Comment:
Even I use `LuigiTestCase`, `TaskClassAmbigiousException: Task 'D' is ambiguous` error is thrown in `test_with_dynamic_dependencies_with_single_worker` with tox. Can you help me on that?